packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -348,8 +348,19 @@ class HotRunner extends ResidentRunner {
     // TODO(aam): Add generator reset logic once we switch to using incremental
     // compiler for full application recompilation on restart.
     final bool updatedDevFS = await _updateDevFS(fullRestart: true);
-    if (!updatedDevFS)
+    if (!updatedDevFS) {
+      for (FlutterDevice device in flutterDevices) {
```

Comment:
The check for device.generator != null could be hoisted outside the loop
if (!updatedevFS) {
  if (device.generator != null) {
    for (FlutterDevice device in flutterDevices) {
      device.generator.reject();
    }
  }
  return new Ope.....
}

Reply:
`device` is the for-loop variable, so the has to be stay inside.